### PR TITLE
SUS-823 Converted abuse filter log to use username lookup

### PR DESCRIPTION
**Pull request by jcellary**
https://wikia-inc.atlassian.net/browse/SUS-823

Added lookup of username instead of relying on local copy for abuse filter log.

@mixth-sense @wladekb @macbre @pwojcik86

### CODE REVIEW

File: extensions/AbuseFilter/special/SpecialAbuseLog.php [hunk 1]
```diff
@@ -185,36 +186,42 @@ function showList() {
...
 
-		if ( $this->mSearchFilter ) {
-			// if the filter is hidden, users who can't view private filters should not be able to find log entries generated by it
-			if ( !AbuseFilter::filterHidden( $this->mSearchFilter ) || AbuseFilterView::canViewPrivate() ) {
-				$conds['afl_filter'] = $this->mSearchFilter;
+		if (!$skipQuery) {
+
+			if( $this->mSearchFilter ) {
+				// if the filter is hidden, users who can't view private filters should not be able to find log entries generated by it
+				if( !AbuseFilter::filterHidden( $this->mSearchFilter ) || AbuseFilterView::canViewPrivate() ) {
```
Comment (WikiaTech) [inline extensions/AbuseFilter/special/SpecialAbuseLog.php]: MAJOR Merge this if statement with the enclosing one. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1066)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 37 issues
* MAJOR 6 major
* MINOR 29 minor
* INFO 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AbuseFilterViewRevert.php#L187](https://github.com/Wikia/app/blob/cd46110bf0f2072f6a02a0c4baf3daf9c5df9ded/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L187): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
1. MAJOR [AbuseFilterViewRevert.php#L188](https://github.com/Wikia/app/blob/cd46110bf0f2072f6a02a0c4baf3daf9c5df9ded/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L188): Reduce this "switch/case" number of lines from 16 to at most 10, for example by extracting code into function. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1151)
1. MAJOR [AbuseFilterViewRevert.php#L209](https://github.com/Wikia/app/blob/cd46110bf0f2072f6a02a0c4baf3daf9c5df9ded/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L209): Reduce this "switch/case" number of lines from 40 to at most 10, for example by extracting code into function. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1151)
1. MAJOR [ApiQueryAbuseLog.php#L37](https://github.com/Wikia/app/blob/cd46110bf0f2072f6a02a0c4baf3daf9c5df9ded/extensions/AbuseFilter/api/ApiQueryAbuseLog.php#L37): The Cyclomatic Complexity of this function "execute" is 25 which is greater than 20 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. MAJOR [ApiQueryAbuseLog.php#L111](https://github.com/Wikia/app/blob/cd46110bf0f2072f6a02a0c4baf3daf9c5df9ded/extensions/AbuseFilter/api/ApiQueryAbuseLog.php#L111): Extract this increment or decrement operator into a dedicated statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS881)
1. MINOR [AbuseFilterViewRevert.php#L32](https://github.com/Wikia/app/blob/cd46110bf0f2072f6a02a0c4baf3daf9c5df9ded/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L32): Define a constant instead of duplicating this literal "wpPeriodStart" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [AbuseFilterViewRevert.php#L34](https://github.com/Wikia/app/blob/cd46110bf0f2072f6a02a0c4baf3daf9c5df9ded/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L34): Define a constant instead of duplicating this literal "wpPeriodEnd" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [AbuseFilterViewRevert.php#L41](https://github.com/Wikia/app/blob/cd46110bf0f2072f6a02a0c4baf3daf9c5df9ded/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L41): Define a constant instead of duplicating this literal "action" 4 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [AbuseFilterViewRevert.php#L63](https://github.com/Wikia/app/blob/cd46110bf0f2072f6a02a0c4baf3daf9c5df9ded/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L63): Define a constant instead of duplicating this literal "actions" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [AbuseFilterViewRevert.php#L70](https://github.com/Wikia/app/blob/cd46110bf0f2072f6a02a0c4baf3daf9c5df9ded/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L70): Define a constant instead of duplicating this literal "userid" 6 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/AbuseFilter/special/SpecialAbuseLog.php (hunk 1 above)
Comment (bognix) [inline extensions/AbuseFilter/special/SpecialAbuseLog.php]: please address comment above

**Comment by WikiaTech** [comment]
SonarQube analysis reported 36 issues
- MAJOR 5 major
- MINOR 29 minor
- INFO 2 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [AbuseFilterViewRevert.php#L187](https://github.com/Wikia/app/blob/ea3809bc123e27f63a49fbb464335522e4892908/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L187): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
2. MAJOR [AbuseFilterViewRevert.php#L188](https://github.com/Wikia/app/blob/ea3809bc123e27f63a49fbb464335522e4892908/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L188): Reduce this "switch/case" number of lines from 16 to at most 10, for example by extracting code into function. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1151)
3. MAJOR [AbuseFilterViewRevert.php#L209](https://github.com/Wikia/app/blob/ea3809bc123e27f63a49fbb464335522e4892908/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L209): Reduce this "switch/case" number of lines from 40 to at most 10, for example by extracting code into function. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1151)
4. MAJOR [ApiQueryAbuseLog.php#L37](https://github.com/Wikia/app/blob/ea3809bc123e27f63a49fbb464335522e4892908/extensions/AbuseFilter/api/ApiQueryAbuseLog.php#L37): The Cyclomatic Complexity of this function "execute" is 25 which is greater than 20 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
5. MAJOR [ApiQueryAbuseLog.php#L111](https://github.com/Wikia/app/blob/ea3809bc123e27f63a49fbb464335522e4892908/extensions/AbuseFilter/api/ApiQueryAbuseLog.php#L111): Extract this increment or decrement operator into a dedicated statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS881)
6. MINOR [AbuseFilterViewRevert.php#L32](https://github.com/Wikia/app/blob/ea3809bc123e27f63a49fbb464335522e4892908/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L32): Define a constant instead of duplicating this literal "wpPeriodStart" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
7. MINOR [AbuseFilterViewRevert.php#L34](https://github.com/Wikia/app/blob/ea3809bc123e27f63a49fbb464335522e4892908/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L34): Define a constant instead of duplicating this literal "wpPeriodEnd" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
8. MINOR [AbuseFilterViewRevert.php#L41](https://github.com/Wikia/app/blob/ea3809bc123e27f63a49fbb464335522e4892908/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L41): Define a constant instead of duplicating this literal "action" 4 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
9. MINOR [AbuseFilterViewRevert.php#L63](https://github.com/Wikia/app/blob/ea3809bc123e27f63a49fbb464335522e4892908/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L63): Define a constant instead of duplicating this literal "actions" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
10. MINOR [AbuseFilterViewRevert.php#L70](https://github.com/Wikia/app/blob/ea3809bc123e27f63a49fbb464335522e4892908/extensions/AbuseFilter/Views/AbuseFilterViewRevert.php#L70): Define a constant instead of duplicating this literal "userid" 6 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Comment by bognix** [comment]
LGTM

**Pull request merged**